### PR TITLE
Add documentation for calendar/fixed intervals

### PR DESCRIPTION
**Pull request by polyfractal**
This adds documentation that was missed in https://github.com/elastic/elasticsearch/pull/33727 :scream: 

I tried to reuse the text about the different intervals and their definitions, but I purged some of the free form text since it was hard to read and didn't really fit with the new parameters.

There is a section that mentions the old `interval` parameter, but only in passing.  Wasn't sure if we wanted to fully document the now-deprecated setting or not.  Seems like it would needlessly complicate the documentation imo.

### CODE REVIEW

File: docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc [hunk 1]
```diff
@@ -10,102 +10,122 @@ that here the interval can be specified using date/time expressions. Time-based
...
-minute that contains a leap-second, which is 2000ms long); supports multiples.
+Calendar-aware intervals understand that daylight savings changes the length
+of specific days, months have different amounts of days, and leap seconds can
+be tacked onto a particular year.
 
-minutes (m) ::
+Fixed intervals are, by contrast, always multiples of SI units and do not change
+based on calendaring context.
+
+Historically these were configured in a single `interval` field, which lead to
```
Comment (talevy) [inline docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc]: can we highlight this historical behavior by boxing it into its own sub-block called #deprecated or something? A user skimming the docs may not catch that this is no longer the behavior and only providing context for those familiar with the old ways.

Another option is to leave these comments in the 7x branch, and remove this historical discussion in 8.0?

File: docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc [hunk 2]
```diff
@@ -10,102 +10,122 @@ that here the interval can be specified using date/time expressions. Time-based
...
+the user would need to specify `24h`.
+
+This behavior has been deprecated in favor of two new, explicit fields: `calendar_interval`
+and `fixed_interval`.
+
+===== Calendar Intervals
+
+Calendar-aware intervals are configured with the `calendar_interval` parameter.
+Calendar intervals can only be specified in "singular" quantities of the unit
+(`1d`, `1M`, etc).  Multiples, such as `2d`, are not supported and will throw an exception.
```
Comment (talevy) [inline docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc]: should we provide more examples hitting this home? I imagine users will hit this exception very quickly. Might be nice to bring them peace by showing this exception

**Review by talevy — COMMENTED** [review]
I tried reading it as a naive user, which was easy because I didn't have to pretend.

I liked it, I thought it was very clear. I just left a few comments

File: docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc [hunk 3]
```diff
@@ -10,102 +10,122 @@ that here the interval can be specified using date/time expressions. Time-based
...
+Calendar-aware intervals understand that daylight savings changes the length
+of specific days, months have different amounts of days, and leap seconds can
+be tacked onto a particular year.
 
-minutes (m) ::
+Fixed intervals are, by contrast, always multiples of SI units and do not change
+based on calendaring context.
+
+Historically these were configured in a single `interval` field, which lead to
+confusing semantics.  Specifying `1d` would be assumed as a calendar-aware time,
```
[the remark this answers is not in the document]
Comment (polyfractal) [reply docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc]: You an @jasontedor are barbarians.

(will change :) )

File: docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc (hunk 1 above)
Comment (polyfractal) [reply docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc]: ++ a callout info block makes sense to me, and probably add a few extra sentences describing the change.

File: docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc (hunk 2 above)
Comment (polyfractal) [reply docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc]: Good point.  I can add an example of fixed intervals choking on units like `month` too, just to cover that side as well.

**Comment by polyfractal** [comment]
Thanks @talevy!  Pushed a commit which addresses the review comments

**Review by talevy — APPROVED** [review]
awesome! love the examples

**Pull request merged**